x-pack/plugins/enterprise_search/server/routes/app_search/engines.test.ts

```diff
@@ -108,6 +108,73 @@ describe('engine routes', () => {
     });
   });
 
+  describe('POST /api/app_search/engines', () => {
+    const AUTH_HEADER = 'Basic 123';
+    const mockRequest = {
+      headers: {
+        authorization: AUTH_HEADER,
+      },
+      body: {
+        name: 'some-engine',
+        language: 'en',
+      },
+    };
+
+    let mockRouter: MockRouter;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockRouter = new MockRouter({
+        method: 'post',
+        path: '/api/app_search/engines',
+        payload: 'body',
+      });
+
+      registerEnginesRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('creates a request handler', () => {
+      mockRouter.callRoute(mockRequest);
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/engines',
+        body: {
+          name: 'some-engine',
+          language: 'en',
+        },
+      });
+    });
+
+    it('passes the body to enterpriseSearchRequestHandler', () => {
+      mockRouter.callRoute({ body: { name: 'some-engine', language: 'en' } });
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/engines',
+        body: { name: 'some-engine', language: 'en' },
+      });
+    });
+
+    describe('validates', () => {
+      it('correctly', () => {
+        const request = { body: { name: 'some-engine', language: 'en' } };
+        mockRouter.shouldValidate(request);
+      });
+
+      it('missing name', () => {
+        const request = { body: { language: 'en' } };
+        mockRouter.shouldThrow(request);
+      });
+
+      it('missing language', () => {
```

Comment:
it's OK if language is missing so I might change this test name to reflect that

```suggestion
      it('optional language', () => {
```